### PR TITLE
Add MSC2604: Parameters for Login Fallback to the spec.

### PR DESCRIPTION
**Pull request by clokep**
This adds #2604 to the specification.

I feel like this should cross reference back to the normal login API, but I couldn't figure out how to do that.

### CODE REVIEW

File: specification/client_server_api.rst [hunk 1]
```diff
@@ -1096,6 +1096,12 @@ This returns an HTML and JavaScript page which can perform the entire login
 process. The page will attempt to call the JavaScript function
 ``window.onLogin`` when login has been successfully completed.
 
+Non-credential related parameters of the login endpoint can be provided as query
+parameters, for example the ``device_id`` or ``initial_device_display_name``.
```
Comment (clokep) [inline specification/client_server_api.rst:1100]: I would have liked to not list these out so we don't need to potentially update them in the future, but it seemed clearer to be explicit.

File: changelogs/client_server/newsfragments/2709.new [hunk 2]
```diff
@@ -0,0 +1 @@
+Document `MSC2604: Parameters for Login Fallback <https://github.com/matrix-org/matrix-doc/pull/2604>`_.
```
Comment (turt2live) [inline changelogs/client_server/newsfragments/2709.new:1]: This isn't a new endpoint, just an added feature. Should be named `2709.feature`.

For common voice as well:
```suggestion
Add a ``device_id`` parameter to login fallback per `MSC2604 <https://github.com/matrix-org/matrix-doc/pull/2604>`_.
```

File: specification/client_server_api.rst (hunk 1 above)
Comment (turt2live) [inline specification/client_server_api.rst:1100]:
```suggestion
Non-credential parameters valid for the ``/login`` endpoint can be provided as query
string parameters here. For example::
```

Comment (turt2live) [reply specification/client_server_api.rst:1100]: bonus points if `/login` becomes a link.

Comment (turt2live) [reply specification/client_server_api.rst:1100]: oh, rationale: we're not going to remember to update this so it's better to just leave it vague and open.

Comment (clokep) [reply specification/client_server_api.rst:1100]: Right. I couldn't figure out how to make it a link anyway. 😢

Comment (clokep) [reply specification/client_server_api.rst]: Oh hmm, something got funky with accepting this one. Did you expect the "These are to be forwarded to the login endpoint during the login process." line to go away?

Comment (clokep) [reply specification/client_server_api.rst]: I ended up removing that line as well, I think that's what you were going for!

Comment (turt2live) [reply specification/client_server_api.rst]: hmm, that line should have survived, I just forgot about it when writing the suggestion - sorry! I'll stuff it back in.

**Comment by clokep** [comment]
Thank you! 👍

**Pull request merged**